web/src/services/Selector.service.ts

```diff
@@ -43,50 +32,34 @@ const getPseudoSelectorString = (pseudoSelector?: TPseudoSelector): string => {
   return selector;
 };
 
+function flattenStructureFilters(structure: Structure): Filter[] {
+  return [
+    ...(structure?.filters || []),
+    ...(structure?.childSelector ? flattenStructureFilters(structure.childSelector) : []),
+  ];
+}
+
 const SelectorService = () => ({
   getSelectorString(selectorList: TSpanSelector[], pseudoSelector?: TPseudoSelector): string {
     return selectorList.length
       ? `span[${getFilters(selectorList).join('  ')}]${getPseudoSelectorString(pseudoSelector)}`
       : '';
   },
 
-  getSpanSelectorList(selectorString: string): TSpanSelector[] {
-    const [matchString] = (selectorString.match(selectorRegex) || []).reverse();
-
-    if (!matchString) return [];
-
-    const selectorList = matchString.split('  ').reduce<TSpanSelector[]>((list, operation) => {
-      if (!operation) return list;
-      const [key, operator, value] = operation.split(operationRegex);
-
-      const spanSelector = {
-        key: key.trim(),
-        operator: operator.trim() as TCompareOperatorSymbol,
-        value: getCleanValue(value.trim()),
-      };
-
-      return list.concat([spanSelector]);
-    }, []);
-
-    return selectorList;
+  getSpanSelectorList(structures: Structure[]): TSpanSelector[] {
+    return structures
+      .flatMap(a => flattenStructureFilters(a))
+      .map(r => ({
+        key: r?.property || '',
+        operator: r?.operator as TCompareOperatorSymbol,
+        value: r?.value || '',
+      }));
   },
 
-  getPseudoSelector(selectorString: string): TPseudoSelector | undefined {
-    const index = selectorString.indexOf(']:');
-    if (index === -1) return;
-
-    const pseudoSelector = selectorString.substring(selectorString.indexOf(']:') + 1);
-
-    if (pseudoSelector.includes('nth_child')) {
-      const [number] = (pseudoSelector.match(nthChildNumberRegex) || []).reverse();
-
-      return {selector: PseudoSelector.NTH, number: Number(number)};
-    }
-
+  getPseudoSelector(structure: Structure[]): TPseudoSelector | undefined {
```

Comment:
Are we covering the possibility of having multiple pseudo selectors?

Reply:
i think we do not use that yet. this functionality would have to be added when implementing the ancestor functionality

Reply:
Ok, let's just not lose track of that

Reply:
I think either way we have to consider the multiple pseudo selectors as I think that's a separated issue from the ancestors

Reply:
i dont think there is multiple pseudo selectors. there is a single pseudo selector for each selector/structure